### DISCUSSION TITLE
Entre etiquetas no se crea el cable.

### DISCUSSION THREAD

**Discussion by Democrito** [FPGAwars/icestudio#817] [Ideas]
Hola Carlos,

Esta Issue que te voy a comentar es una tontería y no hace falta responder a este mensaje.

No se crea cable entre dos etiquetas, eso es todo. Es muy raro hacer esto porque en apariencia no tiene sentido hacerlo, pero mientras hacía un diseño, por simplicidad y de forma momentánea enlacé dos etiquetas y al cabo de un tiempo me di cuenta de lo que sucedía. Cree un ejemplo para asegurarme y eso ya lo confirmó. Adjunto imagen de ejemplo.

Un abrazo!

![Entre etiquetas no hay cables](https://user-images.githubusercontent.com/5473407/126868113-b46554c0-1574-4ff7-bc5e-2657228a3cee.PNG)

**Comment by Obijuan** [comment]
La semántica actual es que las etiquetas sólo se pueden conectar a bloques, y no entre sí  
El poder conectar dos etiquetas entre sí, simulando un cable, es una característica que tiene muchas utilidades. Yo también lo he encontrado necesario

Para solucionarlo temporalmente he añadido el bloque "wire" a la [colección icewires](https://github.com/FPGAwars/iceWires) cuya función es justamente esa: Un bloque que sólo implementa un cable para poder conectar etiquetas

Añadiendo el bloque wire entre las etiquetas xx - yy del ejemplo que muestras, se consigue que funcione como se espera:

![image](https://user-images.githubusercontent.com/1286062/131504611-edbe623a-f0e5-40ed-aa8c-b0db6a25b9ff.png)

También funciona usando buses:

![image](https://user-images.githubusercontent.com/1286062/131505003-d68428c8-5155-4920-aaeb-98a39ef7e809.png)

Estos bloques se encuentra en el **menú Wires** de la colección icesWires

**Comment by Democrito** [comment]
Muy buena idea (la primera vez que lo vi no me di cuenta de ello), además los "wires" permiten mejorar el diseño ya que después de "torcer" muchas veces una conexión comienza a hacer cosas extrañas (en el sentido gráfico). Sea como fuere las etiquetas ha sido un paso crucial para poder diseñar circuitos complejos y me siento enormemente agradecido (gracias Carlos!).

**Comment by cavearr** [comment]
Esta issue creo que está resuelta ¿la damos por cerrada?

**Comment by Democrito** [comment]
Lo acabo de comprobar y todavía no funciona, al menos a mí.

**Comment by Obijuan** [comment]
Todavía no se puede cerrar. Es una feature que hay que implementar. Lo que propuse aquí es una solución temporal

**Comment by TimRudy** [comment]
I don't understand this (English translation). It seems that `xx` can be connected to `yy` with a wire (2 DIFFERENT labels). That is accepted. It is shown visually. But it does not actually connect them?

What's the desired feature for the user?
- Allow that visual connection (no change to current behaviour)
- Make the wire work as a wire
- Maybe create an alert when they are wiring: "Are you sure you want to connect 2 DIFFERENT labels"?

**Comment by cavearr** [comment]
What democritus is referring to is using links between tags to simplify graphical connections as if they were a "hub".

That cannot be done right now, it would be a new feature to implement.
